tests/optimizers/tests_optimizers.py

```diff
@@ -101,10 +101,10 @@ def test_adagrad():
         #trainer.append_task(tasks.PrintVariable("Loss gradient: {}", loss.gradients[param]))
 
         # Monitor the gradient of `loss` w.r.t. to `param`.
-        gparam = tasks.MonitorVariable(loss.gradients[param])
-        trainer.append_task(gparam)
+        gparam = views.MonitorVariable(loss.gradients[param])
+        trainer._graph_updates.update(gparam.updates)
         trainer.train()
 
         # After 30 epochs, param should be around the center and gradients near 0.
         assert_array_almost_equal(param.get_value(), center)
-        assert_array_almost_equal(gparam.value, 0.)
+        assert_array_almost_equal(gparam.var.get_value(), 0.)
```

Comment:
Then use this:
`assert_array_almost_equal(logger[-1][0], 0.)`

I think the logger usage could be improve, it doesn't seem convivial at all.